modules/admin_manual/pages/oidc/index.adoc

```diff
@@ -0,0 +1,162 @@
+= What is OpenID Connect (OIDC)
+:toc:
+:panva-node-oidc-provider-url: https://github.com/panva/node-oidc-provider
+:openid-connect-frontchannel-logout-url: https://openid.net/specs/openid-connect-frontchannel-1_0.html
+
+According to openid.net/connect:
+
+"OpenID Connect 1.0 is a simple identity layer on top of the OAuth 2.0 protocol. It allows Clients to verify the identity of the End-User based on the authentication performed by an Authorization Server, as well as to obtain basic profile information about the End-User in an interoperable and REST-like manner.
+OpenID Connect allows clients of all types, including Web-based, mobile, and JavaScript clients, to request and receive information about authenticated sessions and end-users. The specification suite is extensible, allowing participants to use optional features such as encryption of identity data, discovery of OpenID Providers, and session management, when it makes sense for them."
+-- https://openid.net/connect/, "What is OpenID Connect?"
+
+== Flows within ownCloud 10 server
+
+== Setup (Installation and Configuration)
+
+* xref:configure-owncloud[Configure ownCloud]
+* xref:setup-within-the-openid-provider[Setup Within the OpenID Provider]
+* xref:setup-service-discovery[Setup Service Discovery]
+* xref:how-to-setup-an-idp-for-development-and-test-purposes[How to Setup an IdP for Development and Test Purposes]
+* xref:integration-with-different-idps[Integration with different IdPs]
+
+=== Configure ownCloud
+
+The OpenID integration is configured by adding the below parameters to ownCloud’s configuration file.
+
+NOTE: `provider-url`, `client-id` and `client-secret` are to be taken from the OpenID Provider’s setup.
+`loginButtonName` can be freely chosen, depending on the installation.
+
+[source,php]
+----
+<?php
+$CONFIG = [
```

Comment:
If you look at: https://github.com/owncloud/core/blob/master/config/config.sample.php
you see that there is already a `$CONFIG = array()` block containing all described config parameters.
Does your config block mean that you need to create a additional config block eg at the end or can this config be included in the current one or do you advise creating a own config_open_id.php file.
Imho you can add this to the current config block, and as variant two, create a seperate conig file for open_id...

Reply:
Additional question:
I have not found in this documentation the need installing or enabling any oC app to make OIDC run.
Due to this, I wonder why this config parametrisation is not put into the regular core:config-sample.php file and reference in this description there. That would make much more sense - imho.

Reply:
> If you look at: https://github.com/owncloud/core/blob/master/config/config.sample.php
> you see that there is already a `$CONFIG = array()` block containing all described config parameters.
> Does your config block mean that you need to create a additional config block eg at the end or can this config be included in the current one or do you advise creating a own config_open_id.php file.
> Imho you can add this to the current config block, and as variant two, create a separate conig file for open_id...

Well spotted. I wanted to include it as a standalone reference so that it is as simple for the user to make use of as possible. However, I see how it may also cause unnecessary confusion. I'll update the sample. Secondly, I hadn't updated config.sample.php yet. That may well come later as part of a separate PR:

Reply:
> Additional question:
> I have not found in this documentation the need installing or enabling any oC app to make OIDC run.
> Due to this, I wonder why this config parametrisation is not put into the regular core:config-sample.php file and reference in this description there. That would make much more sense - imho.

I'll have to talk to @pmaier1 about this.

Reply:
I highly recommend to put the config examples of this PR into config.sample.php and reference it. Anything else is unmaintainable and confusing.

Reply:
I'm not saying that I won't do it. I'm just clarifying that it's coming later.

Reply:
The OIDC part on the side of ownCloud will be shipped as an app. The question of putting information into config.sample.php should be answered based on whether we do this also for other apps that implement config.php options.

Reply:
Thanks for the further feedback, @pmaier1.

Reply:
In this case I propose using `config.apps.sample.php` which is the correct location for.

Reply:
Note, because of only being available when the app is loaded and enabled, I propose dropping a note for that, else people trying to look for but dont get it. Config params which are present but not queried do no harm but are „function less“. In case it is a Enterprise only app, this should be noted in this document too.

Reply:
Thanks, @mmattel. In the middle of doing this now.

Reply:
Has now been actioned.